### DISCUSSION TITLE
zfs hard drive replace question

### DISCUSSION THREAD

**Discussion by daviddean-x** [openzfs/zfs#12241] [Q&A]
Dear all,

It's possible to replace one hard drive from raidz2 pool or from a striped mirror pool with an SSD drive?

Or its critical to mix mechanical and flash devices 'cause the speed different? 

Best Regards 
Dan

**Comment by rincebrain** [comment]
You _can_, as far as I know, replace a drive in a pool with any drive that is at least the same capacity and does not have a smaller logical sector size.

Mixing disks of drastically different IO properties in the same vdev may lead to poor performance. It shouldn't break anything, it just won't improve your performance in that configuration.

Usually I believe the recommendation would be to put the mechanical drives in raidz or normal mirror vdevs and put SSDs in L2ARC/(mirrored) ZIL/(no seriously mirrored) special vdevs.

**Comment by daviddean-x** [comment]
Thanks for the fast answer, zil and l2arc is configured well.

So we can safly replace broken hdd with a ssd, in future we like to move the complete pool to ssds or nvme's.

Thx
